From d3ee8c8df5808b63e86f40831f2ba62cbb33d291 Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Wed, 8 Aug 2018 09:18:27 +0800
Subject: [PATCH] postgresql: Use pkg-config for libxml2 detection.

Upstream-Status: Inappropriate [configuration]

xml2-config does not work. Use pkgconfig to set CPPFLAGS and LIBS.

Signed-off-by: Philip Balister <philip@balister.org>

upate to version 10.4
Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 configure.in | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/configure.in b/configure.in
index d9fe56e..3e2cd3a 100644
--- a/configure.in
+++ b/configure.in
@@ -816,19 +816,8 @@ PGAC_ARG_BOOL(with, libxml, no, [build with XML support],
               [AC_DEFINE([USE_LIBXML], 1, [Define to 1 to build with XML support. (--with-libxml)])])
 
 if test "$with_libxml" = yes ; then
-  PGAC_PATH_PROGS(XML2_CONFIG, xml2-config)
-  if test -n "$XML2_CONFIG"; then
-    for pgac_option in `$XML2_CONFIG --cflags`; do
-      case $pgac_option in
-        -I*|-D*) CPPFLAGS="$CPPFLAGS $pgac_option";;
-      esac
-    done
-    for pgac_option in `$XML2_CONFIG --libs`; do
-      case $pgac_option in
-        -L*) LDFLAGS="$LDFLAGS $pgac_option";;
-      esac
-    done
-  fi
+    CPPFLAGS="$CPPFLAGS `pkg-config --short-errors --print-errors --cflags "libxml-2.0" 2>&1`"
+    LIBS="`pkg-config --short-errors --print-errors --libs "libxml-2.0" 2>&1` $LIBS"
 fi
 
 AC_SUBST(with_libxml)
-- 
2.7.4

